### PR TITLE
Zigduino integration

### PR DESCRIPTION
**Pull request by schrein**
Fix integration test with Zigduino

### CODE REVIEW

File: gateway_code/autotest/autotest.py [hunk 1]
```diff
@@ -279,7 +279,7 @@ def auto_tests(self, channel=None, blink=False, flash=False, gps=False):
 
             # switch to DC and configure open node
             self._setup_open_node_connection()
-            time.sleep(1)
+            time.sleep(3)
```
Comment (rienafairefr) [inline gateway_code/autotest/autotest.py]: est-ce que en mettant un time.sleep(2) dans NodeZigduino.setup ça aurait pas le même effet par hasard ,

Comment (rienafairefr) [reply gateway_code/autotest/autotest.py]: A priori tu veux attendre 2s après que la serial redirection du zigduino est démarré, c'est bien ça ?
pour ne pas impacter les autres noeuds tu pourais monkey_patcher serial_redirection.start si tu veux:

        original_redirection_start = self.serial_redirection.start
        def new_redirection_start():
            original_redirection_start()
            time.sleep(2)
        self.serial_redirection.start = new_redirection_start()

(dans ton NodeZigduino.\_\_init\_\_)

**Comment by schrein** [comment]
> Le 12 déc. 2017 à 12:22, Matthieu Berthomé <notifications@github.com> a écrit :
> 
> @rienafairefr commented on this pull request.
> 
> In gateway_code/autotest/autotest.py <https://github.com/iot-lab/iot-lab-gateway/pull/41#discussion_r156338503>:
> 
> > @@ -279,7 +279,7 @@ def auto_tests(self, channel=None, blink=False, flash=False, gps=False):
>  
>              # switch to DC and configure open node
>              self._setup_open_node_connection()
> -            time.sleep(1)
> +            time.sleep(3)
> A priori tu veux attendre 2s après que la serial redirection du zigduino est démarré, c'est bien ça ?
> 
Yes, that’s it. When control_node_serial start DC, it takes 2s in order the TTY is up and running. It would be the same for all Arduino boards with only one TTY for both flash, and serial print (like uno, mega, and co). 

> pour ne pas impacter les autres noeuds tu pourais monkey_patcher serial_redirection.start si tu veux:
> 
>     original_redirection_start = self.serial_redirection.start
>     def new_redirection_start():
>         original_redirection_start()
>         time.sleep(2)
>     self.serial_redirection.start = new_redirection_start()
Ok that’s one possible solution. 

I’m not sure about our priority we should follow :

-  keep the integration test as generic as possible ? (what I tried to do in this PR)

or

- add hardware specific integration test (what @rienafairefr suggests)


@all just tell me the truth, I will follow ! :-)

**Comment by aabadie** [comment]
Looks like something went wrong with the 'monkey-patch' commit, it doesn't appear in the commit history.

**Comment by rienafairefr** [comment]
It's in a PR on Guillaume's fork. 2 possibilities (extend SerialRedirection or monkey-patch)

**Comment by aabadie** [comment]
> It's in a PR on Guillaume's fork

Actually there are 2. Maybe add a link in this PR and let's discuss here the alternatives ?

**Comment by rienafairefr** [comment]
https://github.com/schrein/iot-lab-gateway/pull/2 Extending the SerialRedirection
https://github.com/schrein/iot-lab-gateway/pull/3 Monkey Patching SerialRedicrection.start

**Comment by schrein** [comment]
Ok, let's go further. Tests are failing when connecting for the first time to the serial port of the Zigduino/Arduino controller. Indeed, the default behaviour when open serial port from an arduino controller is to reset the board. They are different solution to avoir this problem, most of them are hardware:

https://playground.arduino.cc/Main/DisablingAutoResetOnSerialConnection

We can also try to disable DTR from serial port, but connection is forwarded by socat, no directly by iot-lab-gateway Python Code.

**Comment by schrein** [comment]
Ok finally just the DTR was in fault. We can ignore the other serial redirection wait issue.

**Comment by aabadie** [comment]
@schrein can you rebase on latest master branch ?
with `upstream` as the iotlab remote repository and `schrein` your remote fork:
```
   git checkout zigduino_intregration
   git pull --rebase upstream master
   git push schrein zigduino_intregration -f
```

Normally the `git pull --rebase` will work without any errors. After that if the CI is green, we can merge and add a Zigduino open node in the CI job.

**Comment by schrein** [comment]
> @schrein <https://github.com/schrein> can you rebase on latest master branch ?

Done.
> Normally the git pull --rebase will work without any errors. After that if the the CI is green, we can merge and add a Zigduino open node in the CI job.
> 
Travis is green thanks to your commit 78c05a. Great !

**Comment by aabadie** [comment]
I'd like to merge this one before moving `master` branch to `release`. Is it ok for you @fsaintma @rienafairefr ?

**Pull request merged**